backend/migrations/003_add_currency_to_groups.py

```
@@ -0,0 +1,116 @@
+        db = client[settings.database_name]
+        groups_collection = db["groups"]
+
+        # Count total groups
+        total_groups = await groups_collection.count_documents({})
+
+        # Count groups with currency
+        groups_with_currency = await groups_collection.count_documents(
+            {"currency": {"$exists": True, "$ne": None}}
+        )
+
+        logger.info(f"\n📊 Verification Results:")
```

Ruff (0.14.11)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)